### PR TITLE
Custom Profile Picture Upload

### PR DESCRIPTION
**Pull request by AnatoleAM**
Adding an endpoint to allow users to upload a custom profile picture. Currently only supports the GIF format

### CODE REVIEW

File: src/server/api/v2/rest/cosmetics/avatar.go [hunk 1]
```diff
@@ -37,88 +28,111 @@ func Avatar(router fiber.Router) {
 
 	router.Get("/avatars", func(c *fiber.Ctx) error {
```
Comment (TroyDota) [inline src/server/api/v2/rest/cosmetics/avatar.go:29]: How is this cached?

Comment (AnatoleAM) [reply src/server/api/v2/rest/cosmetics/avatar.go:29]: The expectation is it's behind Cloudflare (see L129 where the header is set)

File: src/mongo/datastructure/datastructure.go [hunk 2]
```diff
@@ -109,20 +109,20 @@ type User struct {
...
+	YouTubeID            string              `json:"yt_id,omitempty" bson:"yt_id,omitempty"`
+	DisplayName          string              `json:"display_name" bson:"display_name"`
+	Login                string              `json:"login" bson:"login"`
+	BroadcasterType      string              `json:"broadcaster_type" bson:"broadcaster_type"`
+	ProfileImageURL      string              `json:"profile_image_url" bson:"profile_image_url"`
+	OfflineImageURL      string              `json:"offline_image_url" bson:"offline_image_url"`
+	Description          string              `json:"description" bson:"description"`
+	CreatedAt            time.Time           `json:"twitch_created_at" bson:"twitch_created_at"`
+	ViewCount            int32               `json:"view_count" bson:"view_count"`
+	CustomProfileImageID string              `json:"profile_picture_id,omitempty" bson:"profile_picture_id,omitempty"`
```
Comment (TroyDota) [inline src/mongo/datastructure/datastructure.go:122]: the  bson and json tags should reflect the struct tag,
ie. Either change the struct tag to ProfilePictureID or the json/bson tags to custom_profile_image_id

**Review by TroyDota — CHANGES_REQUESTED** [review]

**Comment by TroyDota** [comment]
image
seems good to me!!!

**Pull request merged**